lib/puma/request.rb

```diff
@@ -92,6 +92,7 @@ def handle_request(client, requests)
       # array, we will invoke them when the request is done.
       #
       env[RACK_AFTER_REPLY] ||= []
+      env[RACK_RESPONSE_FINISHED] ||= []
```

Comment:
```suggestion
      response_finished = []
      env[RACK_RESPONSE_FINISHED] = response_finished
```

Do we actually need `||=` here?